internal/planner/scan.go

```diff
@@ -325,35 +325,53 @@ func (p *Planner) Scan(
 type multiScanNode struct {
 	scanNode   *scanNode
 	numReaders int
-	numCalls   int
-
-	lastBool bool
-	lastErr  error
+	nextCount  int
+	initCount  int
+	startCount int
+	closeCount int
+
+	nextResult bool
+	nextErr    error
+	initErr    error
+	startErr   error
+	closeErr   error
 }
 
 func (n *multiScanNode) Init() error {
-	return n.scanNode.Init()
+	n.countAndCall(&n.initCount, func() {
```

Comment:
todo: Please document why this is done - it is not obvious within this PR, and it was already missed previously.  And please do the same for the other funcs changed here too.

Reply:
documented